src/Collector/SuperGlobalsCollector.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SensioLabs\Deptrac\Collector;
+
+use SensioLabs\Deptrac\AstRunner\AstMap;
+use SensioLabs\Deptrac\AstRunner\AstMap\AstClassReference;
+
+final class SuperGlobalsCollector implements CollectorInterface
+{
+    public function getType(): string
+    {
+        return 'superGlobals';
+    }
+
+    public function satisfy(array $configuration, AstClassReference $astClassReference, AstMap $astMap, Registry $collectorRegistry): bool
+    {
+        $fileReference = $astClassReference->getFileReference();
+        $source = file_get_contents($fileReference->getFilepath());
+
+        return strpos($source, '_GET') !== false;
```

Comment:
I just do a naiv case-sensitive check for the `_GET` super global.

this works because the names of the super global variables are reserved and are case sensitive.
of course this could also be achieved with the power of the AST but depends on some infrastructure which does not exit yet - see e.g. https://github.com/qossmic/deptrac/issues/582

using this very simple approach, I feel we could get this collectors' satisfy-method implemented in 20-30 lines of code, while supporting all known superglobals.

Reply:
I am afraid this approach does not work on so many levels. There are plenty of places that you can use `_GET` string that would not mean you are accessing a superglobal. Just from the top of my head:
```php

namespace My_GET;

class HELLO_GET
{
   public function func_GET(): string
   {
     $var_GET = '_GET';
     return $var_GET;
   }
}